### PR TITLE
hcert_spec.md etc / based on v1-final branch

### PR DESCRIPTION
**Pull request by gabywh**
Based on branch v1-final:

1. lots of typos + grammar fixed
2. optical format: QR code agreed by EHN as the only format, so I removed refs to Aztec
3. compression scheme: as per my earlier email, I modded the text here so that compression can be detected at runtime, with zlib being the default go-to compression mechanism.

Gaby.

### CODE REVIEW

File: hcert_spec.md [hunk 1]
```diff
@@ -119,33 +119,30 @@ If the transfer of the HCERT from the Issuer to the holder is based on a present
 
 ### Barcode
 
-To lower size and to improve speed and reliability in the reading process of the HCERT, the CWT SHALL be compressed using ZLIB ([RFC 1950](https://tools.ietf.org/html/rfc1950)) and the Deflate compression mechanism in the format defined in ([RFC 1951](https://tools.ietf.org/html/rfc1951)). 
+#### Payload (CWT) Compression
 
-Verifiers MUST check of the presence of a valid ZLIB/Deflate header (0x78, 0xDA) - or proceed without this step when absent.
+To lower size and to improve speed and reliability in the reading process of the HCERT, the CWT MAY be compressed. The compression algorithm used shall be detected at runtime (e.g. by interrogating the file header). If no particular constraints are present, the recommended compression mechanism is ZLIB ([RFC 1950](https://tools.ietf.org/html/rfc1950)) and the corresponding deflate compression mechanism as per ([RFC 1951](https://tools.ietf.org/html/rfc1951)).
```
Comment (jschlyter) [inline hcert_spec.md:124]: "the RECOMMENDED compression mechanism" to stronger suggest that it is probably a bad idea to use something completely different.

File: hcert_spec.md [hunk 2]
```diff
@@ -119,33 +119,30 @@ If the transfer of the HCERT from the Issuer to the holder is based on a present
...
+#### Payload (CWT) Compression
 
-Verifiers MUST check of the presence of a valid ZLIB/Deflate header (0x78, 0xDA) - or proceed without this step when absent.
+To lower size and to improve speed and reliability in the reading process of the HCERT, the CWT MAY be compressed. The compression algorithm used shall be detected at runtime (e.g. by interrogating the file header). If no particular constraints are present, the recommended compression mechanism is ZLIB ([RFC 1950](https://tools.ietf.org/html/rfc1950)) and the corresponding deflate compression mechanism as per ([RFC 1951](https://tools.ietf.org/html/rfc1951)). 
 
-In order to better handle legacy equipment designed to operate on ASCII payloads, the compressed CWT is encoded as ASCII using [Base45](https://datatracker.ietf.org/doc/draft-faltstrom-base45) before encoded into a barcode.
+Verifiers MUST check of the presence of a valid compression header. The recommended ZLIB/Deflate header begins with the bytes (0x78, 0xDA). If no
```
Comment (jschlyter) [inline hcert_spec.md:126]: It is important to note that there are other compression headers value. I'm a bit worried that some implementors of verifiers will _just_ check of 0x78 0xDA, when say 0x78 0x9C is perfectly fine. Not sure how we rephrase this though.

**Review by jschlyter — CHANGES_REQUESTED** [review]

File: hcert_spec.md [hunk 3]
```diff
@@ -46,13 +46,13 @@ The integrity and authenticity of origin of HCERT data, the CWT MUST be verifiab
 
 - Protected Header
   - Signature Algorithm (`alg`, label 1)
-  - Key Identifier (`kid`, label 4)
+  - Key Identifier (`KID`, label 4)
```
[the remark this answers is not in the document]
Comment (dirkx) [reply hcert_spec.md:49]: fixed.

File: hcert_spec.md (hunk 1 above)
Comment (dirkx) [reply hcert_spec.md:123]: fixed.

File: hcert_spec.md (hunk 2 above)
Comment (dirkx) [reply hcert_spec.md:125]: fixed.

**Pull request merged**